src/App.jsx

```diff
@@ -14,45 +22,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELDS.length:
+          return good1[sortField] - good2[sortField];
+
+        case SORT_FIELDS.name:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setReverseField] = useState();
+  const sortedOrReversed = sortField || isReversed;
+  const visibleGoods
+    = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
+  const {
+    name,
+    length,
+  } = SORT_FIELDS;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          className={cn(
+            'button is-info',
+            { 'is-light': sortField !== name },
+          )}
+          onClick={() => setSortField(name)}
+          type="button"
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          className={cn(
+            'button is-success',
+            { 'is-light':
+            sortField !== length },
+          )}
+          onClick={() => setSortField(length)}
+          type="button"
+        >
+          Sort by length
+        </button>
```

Comment:
In future, when you gonna use typescript, don't forget that you can iterate through 'enum' of sort type options, and render these buttons in easier way